cartesi-rollups/development/migration.md

```diff
@@ -1,44 +1,114 @@
 
-1. If installed using Homebrew:
+Epoch closure is now determined by the `CARTESI_EPOCH_LENGTH` environment variable (in blocks) instead of `CARTESI_EPOCH_DURATION` (in seconds).
 
-	```bash
-	brew uninstall sunodo
-	```
+```plaintext
+CARTESI_EPOCH_LENGTH = CARTESI_EPOCH_DURATION / BLOCK_TIME
+```
 
-2. If installed using NPM:
-	```bash
-	npm uninstall -g sunodo
-	```
+Where `BLOCK_TIME` is the time to generate a block in the target network.
 
-## Install Cartesi CLI
+**Example**: If a block is generated every 12 seconds and `CARTESI_EPOCH_DURATION` is set to 86400 seconds (24 hours), `CARTESI_EPOCH_LENGTH = 86400 / 12 = 7200`
 
-1. Install with Homebrew:
+### Option 1: Redeploy all contracts
 
-	```shell
-	brew install cartesi/tap/cartesi
-	```
+Redeploy all contracts and your application with the new configuration.
 
-2. Install with NPM:
+Refer to the [self-hosted deployment guide](../deployment/self-hosted.md) for detailed deployment instructions.
 
-	```shell
-	npm install -g @cartesi/cli
-	```
+:::caution
+Redeploying creates a new application instance. All previous inputs, outputs, claims, and funds locked in the application contract will remain associated with the old application address.
+:::
 
+### Option 2: Replace Application's History
 
-## Update Project Configuration
+This process allows inputs, outputs, and locked funds to remain unchanged but is more involved.
+
+:::note
+A new _History_ will have no claims. Upon restarting the node with a new _History_, previous claims will be resubmitted, incurring additional costs for the _Authority_ owner.
+:::
+
+:::caution
+Instances of the [_History_](https://github.com/cartesi/rollups-contracts/blob/v1.2.0/onchain/rollups/contracts/history/History.sol) contract from [rollups-contracts v1.2.0](https://github.com/cartesi/rollups-contracts/releases/tag/v1.2.0) may be used simultaneously by several applications through their associated _Authority_ instance.
+Application owners must consider that and exercise care when performing the steps listed below.
+:::
+
+It's recommended to use the deterministic deployment functions available in the Rollups contracts.
+
+#### 1. Define the environment variables
+
+- `SALT`: A random 32-byte value for the deterministic deployment functions.
+- `RPC_URL`: The RPC endpoint to be used.
+- `MNEMONIC`: The mnemonic phrase for the Authority owner's wallet (other wallet options may be used).
+- `HISTORY_FACTORY_ADDRESS`: The address of a valid HistoryFactory instance.
+- `AUTHORITY_ADDRESS`: The address of the Authority instance used by the application.
+
+  :::note environment variables
+  A `HistoryFactory` is deployed at `0x1f158b5320BBf677FdA89F9a438df99BbE560A26` for all supported networks, including Ethereum, Optimism, Arbitrum, Base, and their respective Sepolia-based testnets.
+  :::
+
+#### 2. Instantiate a New _History_
```

Comment:
@masiedu4 did we have anyone from core reviewing this?

Reply:
Yes JJ thoroughly did review.